src/kanata/mod.rs

```diff
@@ -1014,14 +1014,34 @@ impl Kanata {
                             log::debug!("on-press: sleeping for {delay} ms");
                             std::thread::sleep(std::time::Duration::from_millis((*delay).into()));
                         }
-                        CustomAction::SequenceLeader => {
+                        CustomAction::SequenceCancel => {
+                            if !self.sequence_state.is_none() 
+                            {
+                                log::debug!("restarting sequence");
+                                let state = self.sequence_state.as_ref().unwrap();
+                                match self.sequence_input_mode {
```

Comment:
Thought: I believe this is duplicated code from the other instance of ending a sequence, is that right? Might be good to see if it can be deduplicated by adding a new function. The new function should probably be standalone and accept references to state and kbd_out, since if it accepted self that might cause borrowing issues since the compiler might not be able to determine that they are disjoint borrows.

Reply:
Yes this is just duplicated code. I will put that into a function.